src/main/java/fr/sncf/osrd/cbtc/CBTCNavigatePhase.java

```diff
@@ -0,0 +1,170 @@
+package fr.sncf.osrd.cbtc;
+
+import java.util.ArrayList;
+import java.util.List;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import fr.sncf.osrd.infra.StopActionPoint;
+import fr.sncf.osrd.simulation.Simulation;
+import fr.sncf.osrd.simulation.SimulationError;
+import fr.sncf.osrd.simulation.TimelineEvent;
+import fr.sncf.osrd.speedcontroller.SpeedController;
+import fr.sncf.osrd.train.Interaction;
+import fr.sncf.osrd.train.Train;
+import fr.sncf.osrd.train.TrainPath;
+import fr.sncf.osrd.train.TrainSchedule;
+import fr.sncf.osrd.train.TrainState;
+import fr.sncf.osrd.train.TrainStatus;
+import fr.sncf.osrd.train.TrainStop;
+import fr.sncf.osrd.train.events.TrainMoveEvent;
+import fr.sncf.osrd.train.events.TrainReachesActionPoint;
+import fr.sncf.osrd.train.phases.NavigatePhase;
+import fr.sncf.osrd.train.phases.NavigatePhaseState;
+import fr.sncf.osrd.utils.TrackSectionLocation;
+
+/**
+ * Navigation phase to be used when the train is CBTC
+ */
+public final class CBTCNavigatePhase extends NavigatePhase {
+
+    /**
+     * Creates a CBTC navigation phase with an already determined path and interaction path.
+     * 
+     * @param startLocation the location of the beginning of the phase
+     * @param endLocation the location of the end of the phase
+     * @param interactionsPath the list of interaction points crossed during the phase
+     * @param expectedPath the path that the train is supposed to follow during the phase
+     * @return The new CBTC navigation phase.
+     */
+    private CBTCNavigatePhase(TrackSectionLocation startLocation,
+            TrackSectionLocation endLocation, ArrayList<Interaction> interactionsPath, TrainPath expectedPath) {
+        super(startLocation, endLocation, interactionsPath, expectedPath);
+    }
+
+    /**
+     * Creates a new CBTC navigation phase from an already determined path
+     * 
+     * @param driverSightDistance the sight distance of the driver 
+     * @param startLocation the location of the beginning of the phase
+     * @param endLocation the location of the end of the phase
+     * @param expectedPath the path that the train is supposed to follow during the phase
+     * @param stops the list of stopping points that the train will cross during the phase
+     * @return The new CBTC navigation phase.
+     */
+    public static CBTCNavigatePhase from(double driverSightDistance, TrackSectionLocation startLocation,
+            TrackSectionLocation endLocation, TrainPath expectedPath, List<TrainStop> stops) {
+        if (stops == null)
+            stops = new ArrayList<>();
+
+        var actionPointPath = trackSectionToActionPointPath(driverSightDistance, expectedPath, startLocation,
+                endLocation, expectedPath.trackSectionPath);
+        addStopInteractions(actionPointPath, startLocation, endLocation, expectedPath, stops);
+        return new CBTCNavigatePhase(startLocation, endLocation, actionPointPath, expectedPath);
+    }
```

Comment:
I have looked a bit into how this would be done and it's not as easy as I thought. It's possible but the factorized version is actually less "clean" than this, so let's not bother for a few duplicated lines. We can reconsider if this method eventually becomes more complex.